app/assets/javascripts/notifications.js

```diff
@@ -238,8 +238,10 @@ var onReadyNotifications = function() {
     emptyNotificationsList();
     showNotificationsNone();
     changeTitle(0);
+    $('#notifications #notifications_text .align_right h1#clear_notifcations').hide();
```

Comment:
You could possibly get away with just using `$(#notifications #clear_notifications).hide()`